app/assets/javascripts/test/tracing-view-screenshot-tests/dataset_rendering.screenshot.js

```diff
@@ -0,0 +1,124 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+const test = require("ava");
+const puppeteer = require("puppeteer");
+const pixelmatch = require("pixelmatch");
+const { PNG } = require("pngjs");
+const fs = require("fs");
+const path = require("path");
+const { screenshotDataset, DEV_AUTH_TOKEN } = require("./dataset_rendering_helpers");
+
+process.on("unhandledRejection", (err, promise) => {
+  console.error("Unhandled rejection (promise: ", promise, ", reason: ", err, ").");
+});
+
+const BASE_PATH = path.join(__dirname, "../screenshots");
+
+let BRANCH = "master";
+if (!process.env.BRANCH) {
+  console.warn(
+    "[Warning] No branch specified, assuming branch master. If you want to specify a branch, prepend BRANCH=<branchname> to the command.",
+  );
+} else {
+  BRANCH = process.env.BRANCH;
+  console.log(`[Info] Executing tests on branch ${BRANCH}.`);
+}
+
+function compareScreenshot(screenshotBuffer, name, width, height) {
+  return new Promise(resolve => {
+    let readFiles = 0;
+    const existingScreenshot = fs
+      .createReadStream(`${BASE_PATH}/${name}.png`)
+      // If there is no existing screenshot, save the current one
+      .on("error", saveScreenshot)
+      .pipe(new PNG())
+      .on("parsed", doneReading);
+
+    const newScreenshot = new PNG({ width, height });
+    newScreenshot.parse(screenshotBuffer, doneReading);
+
+    function doneReading() {
+      if (++readFiles < 2) return;
+      const diff = new PNG({ width, height });
+
+      const pixelErrors = pixelmatch(
+        existingScreenshot.data,
+        newScreenshot.data,
+        diff.data,
+        existingScreenshot.width,
+        existingScreenshot.height,
+        { threshold: 0.0 },
+      );
+
+      if (pixelErrors > 0) {
+        diff
+          .pack()
+          .pipe(fs.createWriteStream(`${BASE_PATH}/${name}-diff.png`))
```

Comment:
Would it make sense to also dump the new screenshot to the disk? Then, updating the screenshot would be a matter of looking at the diff and the overwriting the old screenshot with the new one. Then, the suite doesn't need to be run again.

Reply:
I think that's a good idea. I'll put those and the diff pngs in a subdirectory and put the path into the .gitignore I'd suggest.

Reply:
Yes, sounds good!

Reply:
I opted to put them in the same directory and name them `<datasetName>.{diff|new}.png` to allow for fast switching between old and new screenshots. The `.{old|new}.png` files are then ignored by git.